documentation/docs/steps/transportRequestUploadCTS.md

```diff
@@ -0,0 +1,173 @@
+    dockerImage: 'my/fiori-node'
+    deployToolDependencies: []
+```
+
+## Building the SAPUI5 Application
+
+Build your SAPUI5 application with the build command of the SAPUI5 toolset and use the step [npmExecuteScripts](npmExecuteScripts.md) to run the build command, proceeding as follows:
+
+1. Configure the step in the `package.json` file of your project
+
+   ```json
+   {
+      ...
+      "scripts": {
+         "start": "ui5 serve",
+         "test": "npm run lint",
+         "build": "ui5 build --clean-dest",
+         ...
+      },
+      "dependencies": {},
+      "devDependencies": {
+         "@ui5/cli": "^2.11.2",
+         ...
+      }
+   }
+   ```
+
+1. Setup the execution step in the pipeline
+
+   ```groovy
+   stage('Build') {
+      npmExecuteScripts(script: this, runScripts: ['build'])
+   }
+   ```
+
+**Note:** Do not use the `mtaBuild` step. The `mta` is dedicated to the SAP Cloud Platform. It does neither create the expected `dist` folder nor the compliant content.
+
+## Uploading the SAPUI5 Application
+
+The Fiori toolset uses the [ODATA service](https://ui5.sap.com/#/topic/a883327a82ef4cc792f3c1e7b7a48de8) to upload your UI5 application to the SAPUI5 ABAP Repository.
+The access is controlled by Basic Authentication (user/password based authentication).
+
+**Note:** Do not upload to SAP BTP. The SAP Business Technology Platform does not support Basic Authentication.
+
+**Note:** Use an HTTPS endpoint to ensure the encryption of your credentials.
+
+## Specifying the Transport Request
+
+The target of the upload is a transport request, identified by an identifier (ID).
+
+The step `transportRequestUploadCTS` allows you to set the ID by parameter.
+
+Alternatively, you can pass the ID through the `commonPipelineEnvironment`.
+For example, by performing a step that generates the ID or obtains it differently.
+See [transportRequestReqIDFromGit](transportRequestReqIDFromGit.md).
+
+### Adding a Parameter
+
+A parameterized pipeline allows you to specify the ID with the launch of the build instead of entering it statically into the pipeline.
+
+```groovy
+transportRequestUploadCTS(
+    script: this,
+    transportRequestId: ${TRANSPORT_REQUEST_ID},
+    ...
+)
+```
+
+The Jenkins pipeline `input` step allows you to specify the ID at runtime of the pipeline.
+
+```groovy
+def ids = input( message: "Upload?",
+    parameters: [
+        string(name: 'TRANSPORT_REQUEST_ID',description: 'Transport Request ID')
+    ]
+)
+
+transportRequestUploadCTS(
+    script:this,
+    transportRequestId: ids['TRANSPORT_REQUEST_ID'],
+    ...
+)
+```
+
+## Common Pipeline Environment
+
+Use the step [transportRequestReqIDFromGit](transportRequestReqIDFromGit.md) to obtain the  `transportRequestId` value from your Git commit messages.
+
+This step extracts the ID from the commit messages of your project repository and enters it into the `commonPipelineEnvironment`, in turn, the upload step `transportRequestUploadCTS` picks it up from there.
```

Comment:
This step extracts the ID from the commit messages of your project repository and enters it in `commonPipelineEnvironment`. In turn, the upload step `transportRequestUploadCTS` picks it up from there.